airflow/utils/log/task_context_logger.py

```diff
@@ -57,7 +88,7 @@ def __init__(self, component_name: str, call_site_logger: Logger | None = None):
     def _should_enable(self) -> bool:
         if not conf.getboolean("logging", "enable_task_context_logger"):
             return False
-        if not getattr(self.task_handler, "supports_task_context_logging", False):
+        if not self.task_handler:
```

Comment:
This is just a simplification

Reply:
is it though?  isn't it needed for backcompat?

Reply:
oh, because it's none in that case; redundant logic yes?

Reply:
Yes